backend/dal/Services/ResearchFileService.cs

```diff
@@ -0,0 +1,61 @@
+using System.Collections.Generic;
+using System.Security.Claims;
+using Microsoft.Extensions.Logging;
+using Pims.Dal.Entities;
+using Pims.Dal.Helpers.Extensions;
+using Pims.Dal.Repositories;
+using Pims.Dal.Security;
+
+namespace Pims.Dal.Services
+{
+    public class ResearchFileService : IResearchFileService
+    {
+        private readonly ClaimsPrincipal _user;
+        private readonly ILogger _logger;
+        private readonly IResearchFileRepository _researchFileRepository;
+        private readonly IPropertyRepository _propertyRepository;
+
+        public ResearchFileService(ClaimsPrincipal user, ILogger<ResearchFileService> logger, IResearchFileRepository researchFileRepository, IPropertyRepository propertyRepository)
+        {
+            _user = user;
+            _logger = logger;
+            _researchFileRepository = researchFileRepository;
+            _propertyRepository = propertyRepository;
+        }
+
+        public PimsResearchFile Add(PimsResearchFile researchFile)
+        {
+            _logger.LogInformation("Adding research file...");
+            _user.ThrowIfNotAuthorized(Permissions.ResearchFileAdd);
+            researchFile.ResearchFileStatusTypeCode = "ACTIVE";
+
+            foreach (var researchProperty in researchFile.PimsPropertyResearchFiles)
+            {
+                //var currentProperty = researchProperty.Property;
+                var pid = researchProperty.Property.Pid.Value;
+                try
+                {
+                    var foundProperty = _propertyRepository.GetByPid(pid);
+                    researchProperty.Property = foundProperty;
+                }
+                catch (KeyNotFoundException e)
```

Comment:
IMO the result of a query not finding anything should return an empty object, null, or a wrapped result with a payload. Throwing an exception indicates that it should ALLWAYS find something, that is not the case here.

Reply:
Sure, this is a nice side effect of having a wrapping service layer, we can hide internal db errors and wrap them with appropriate business logic.